Modules/Microsoft365DSC/DSCResources/MSFT_AADApplication/MSFT_AADApplication.psm1

```diff
@@ -386,6 +409,44 @@ function Set-TargetResource
         $currentParameters.Remove("Homepage") | Out-Null
     }
 
+    if ($Ensure -ne 'Absent') {
+        $desiredOwnersValue = @()
+        if ($Owners.Length -gt 0)
+        {
+            $desiredOwnersValue = $Owners
+        }
+        if (!$backCurrentOwners)
+        {
+            $backCurrentOwners = @()
+        }
+        $ownersDiff = Compare-Object -ReferenceObject $backCurrentOwners -DifferenceObject $desiredOwnersValue
+        foreach ($diff in $ownersDiff)
+        {
+            if ($diff.InputObject.Contains('@')) {
+                $OwnerId = $(Get-MgUser -UserId $diff.InputObject).Id
```

Comment:
Please have a look at this recent change in another resource:  
https://github.com/microso95ft/Microsoft365DSC/blob/17984cf89f4c13c591e1b935564e060f2be81273/Modules/Microsoft365DSC/DSCResources/MSFT_AADConditionalAccessPolicy/MSFT_AADConditionalAccessPolicy.psm1#L256

The error action was introduced due to some issues with non-existing users.

Reply:
I always use exceptions whenever possible, but just started looking at this project, for which I'll need in my $DAYJOB, and didn't want to drift too much of the code already in place so I basically started with a copy from the AADGroup resource. That being said in the new commit it now checks if objects exist enclosed in exceptions before proceeding, but that being said it shows that AADGroup also has the same problem. If this one now is OK I can apply the same fix there.

Reply:
Good catch. Handling errors would be the preferred solution :-)

Reply:
I was on the fence on that one, but I agree it should stop to give the admin the chance to inspect what went wrong in the first place, I've updated it.